### ISSUE TITLE
Save canvas created with the images

### ISSUE THREAD

**Issue by jothikannan** [kangax/fabric.js#1964]
i want to store the canvas generated with the fabric.js, i have canvas as like below in my page

`<div class="canvas-container" style="width: 600px; height: 600px; position: relative; -moz-user-select: none;">
    <canvas height="600" width="600" id="design-stage" class="lower-canvas" style="position: absolute; width: 600px; height: 600px; left: 0px; top: 0px; -moz-user-select: none;"></canvas>
    <canvas class="upper-canvas " style="position: absolute; width: 600px; height: 600px; left: 0px; top: 0px; -moz-user-select: none; cursor: default;" width="600" height="600"></canvas>
    </div>`

as said in the documentation i tried `serialization`  to save the canvas as json, but am unable to get the objects in the canvas it always returns empty as follows

`{"objects":[],"background":""}`

and my code to generate the json is follow

`var canvas = new fabric.Canvas('design-stage');
console.log(canvas);
var json = JSON.stringify( canvas.toJSON()  );
console.log(json);`

**Comment by jagi** [comment]
Do you have any objects on the canvas?

**Comment by jothikannan** [comment]
Hi @jagi 

Am not created objects using the functions in the fabric.js, i have loaded the svg images from a directory and added those images into the canvas by positioning the images.

**Comment by jothikannan** [comment]
My canvas have the look like the below attached 
![honda](https://cloud.githubusercontent.com/assets/4100047/6105902/6b0b4652-b082-11e4-9636-d50bc98de5ed.png)
image

**Comment by jagi** [comment]
How did you position images on the canvas? Show the code. You can't serialize fabric canvas without adding image objects to the canvas using `add()` function.

**Comment by asturur** [comment]
if you positioned them just with css posotioning, or by contex.drawimage, fabric instance of  canvas does not know anything about theyr exisence. so you get 0 objects.

**Comment by jothikannan** [comment]
Hi all,

i used fabric.Image.fromURL to set the svg images on the canvas,  and my function is follows

 `self._loadItemImage = function (url) {
        fabric.Image.fromURL(url, function (img) {
            var w = img.width;
            var h = img.height;
            var size = self._resizeImage(w, h, self._width - 100, self._height - 100);
            img.set({
                left: self._width / 2,
                top: self._height / 2,
                originX: 'center',
                originY: 'center',
                hasBorders: false,
                hasControls: false,
                hasRotatingPoint: false,
                selectable: false,
                scaleX: 0.5,
                scaleY: 0.5
            });
            self._motorImg = img;
            self._drawResources();
        });
    };`

**Comment by asturur** [comment]
you are missing canvas.add(img) on the callback function of loadFromURL.
in that case fabric will manage such images.
i really do not know what drawResources does, you should not need it.

**Comment by jothikannan** [comment]
Hi @asturur 

in drawResources i have added the image to the canvas as  `self._canvas.add(self._motorImg);`

**Comment by asturur** [comment]
ok so this is unclear.
we have fragmented information on the ongoing process. i notice now in your code that you do new fabric canvas one line below generating the serialization. the object adding has to happen after the canvas creation and before the serialization, sorry to point out something that maybe is obvious for you.

are you adding the object on the same instance where you added objects or are you creating a new canvas before serialization? 

in other words is self._canvas the same as canvas you later serialize? i  m not speaking about same html element but same fabric instance.

**Comment by jothikannan** [comment]
can i have your email address please ?, still my tool is beta, so i don't want to publish my website link as open, so i can share you the link, there you can see what is happen.

**Comment by asturur** [comment]
my address should be visible in my github profile, i won t have time to check it before wednesday. so you may be able to solve in other way.

**Comment by kangax** [comment]
Please ask questions on google group — https://groups.google.com/forum/#!forum/fabricjs
